workers/unity/Packages/com.improbable.gdk.core/Subscriptions/EntityGameObjectLinker.cs

```diff
@@ -129,10 +134,13 @@ public void UnlinkGameObjectFromEntity(EntityId entityId, GameObject gameObject)
             }
 
             var linkComponent = gameObject.GetComponent<LinkedEntityComponent>();
-            // todo check if this can happen - right now I think it can on application quit
             if (linkComponent != null)
             {
-                linkComponent.Invalidate();
+                linkComponent.IsValid = false;
+                linkComponent.EntityId = new EntityId(0);
+                linkComponent.World = null;
+                linkComponent.Worker = null;
+                linkComponent.Linker = null;
```

Comment:
This and the linking action feels like an:

`Init()` and `Deinit()` pair of methods on the linkComponent.

Reply:
Discussed offline. Agreed but will merge this in for now.